trame_gwc/widgets/gwc.py

```diff
@@ -0,0 +1,346 @@
+    def __init__(self, _elem_name, children=None, **kwargs):
+        super().__init__(_elem_name, children, **kwargs)
+        if self.server:
+            self.server.enable_module(module)
+
+class GirderAccessControl(HtmlElement):
+    """
+        Wraps GirderAccessControl Vue component
+
+        :param model: Object (required)
+        :param has_permission: Boolean (default: False)
+
+    """
+
+    def __init__(self, children=None, **kwargs):
+        super().__init__(
+            "girder-access-control",
+            children,
+            **kwargs,
+        )
+
+        self._attr_names += [
+            "model",
+            ("has_permission", "hasPermission"),
+        ]
+
+
+class GirderAuthentication(HtmlElement):
+    """
+        Wraps GirderProviderAuthentication Vue component
+
+        :param register: Boolean (default: False)
+        :param oauth: Boolean (default: False) 
+        :param forgot_password_url: String (default: null)
+        :param forgot_password_route: [Object, String] (default: null)
+        :param force_otp: Boolean (default: False)
+        :param hide_forgot_password: Boolean (default: False)
+
+    """
+
+    def __init__(self, children=None, **kwargs):
+        super().__init__(
+            "girder-authentication",
+            children,
+            **kwargs,
+        )
+
+        self._attr_names += [
+            "register",
+            "oauth",
+            ("forgot_password_url", "forgotPasswordUrl"),
+            ("forgot_password_route", "forgotPasswordRoute"),
+            ("force_otp", "forceOtp"),
+            ("hide_forgot_password", "hideForgotPassword"),
+        ]
+
+
+class GirderBreadcrumb(HtmlElement):
+    """
+        Wraps GirderBreadcrumb Vue component
+
+        :param location: Object (required)
+        :param readonly: Boolean (default: False) 
+        :param append: Array (default: [])
+        :param root_location_disabled: Boolean (default: False)
+    """
+
+    def __init__(self, children=None, **kwargs):
+        super().__init__(
+            "girder-breadcrumb",
+            children,
+            **kwargs,
+        )
+
+        self._attr_names += [
+            "location",
+            "readonly",
+            "append",
+            ("root_location_disabled", "rootLocationDisabled"),
+        ]
+
+
+class GirderDataDetails(HtmlElement):
+    """
+        Wraps GirderDataDetails Vue component
+
+        :param value: Array (required)
+        :param info_keys: Array (default: DefaultInfoKeys)
+        :param action_keys: Array (default: DefaultActionKeys)
+    """
+
+    def __init__(self, children=None, **kwargs):
+        super().__init__(
+            "girder-data-details",
+            children,
+            **kwargs,
+        )
+
+        self._attr_names += [
+            ("value", "value"),
+            ("info_keys", "infoKeys"),
+            ("action_keys", "actionKeys"),
+        ]
+
+        self._event_names += [
+            "action",
```

Comment:
@bourdaisj  @jourdain when clicking on the associated button, there is an error because trame tries to serialize the action. And the [action](https://github.com/girder/girder_web_components/blob/master/src/components/DataDetails.vue#L183) contains an asynchronous function, hence a serialization error (in `encode()`).
Is it expected ?

Reply:
I think the issue is the return value of your function, not its async nature.

Reply:
I think the issue is the return value of your function, not its async nature.

Reply:
@jourdain if it is an async function, doesn't it means that it returns a `Promise`?
If the return value being a promise is not a problem, do you mean the problem comes from the "type of the value resolved by the promise" ?

Reply:
yes, that is because that value is not serializable. But in that case, it seems that you should not return anything. (or a promise that resolve to None)